### PR TITLE
Deleting aliases should update room

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: tests/30rooms/05aliases.pl [hunk 1]
```diff
@@ -147,3 +147,185 @@
          )->main::expect_http_403;
       })
    };
+
+
```
Comment (leonerd) [inline tests/30rooms/05aliases.pl]: Extraneous blank line

File: tests/30rooms/05aliases.pl [hunk 2]
```diff
@@ -147,3 +147,185 @@
...
+         do_request_json_for( $creator,
+            method => "PUT",
+            uri    => "/r0/directory/room/$room_alias",
+
+            content => { room_id => $room_id },
+         )
+      })->then( sub {
+         matrix_put_room_state( $creator, $room_id,
+            type => "m.room.canonical_alias",
```
Comment (leonerd) [inline tests/30rooms/05aliases.pl]: Align the `=>` for consistency

File: tests/30rooms/05aliases.pl [hunk 3]
```diff
@@ -147,3 +147,185 @@
...
+            type      => "m.room.aliases",
+            state_key => $server_name,
+         )
+      })->then( sub {
+         my ( $body ) = @_;
+
+         assert_json_keys( $body, qw( aliases ) );
+         assert_json_list( my $aliases = $body->{aliases} );
+
+         $room_alias ~~ @$aliases or die "Expected alias to be in list";
```
Comment (leonerd) [inline tests/30rooms/05aliases.pl]: Smartmatch is evil and should be taken out and shot.

Instead I'd suggest an `any` call:

    any { $_ eq $room_alias } @$aliases or
       die "Expected ..."

File: tests/30rooms/05aliases.pl [hunk 4]
```diff
@@ -147,3 +147,185 @@
...
+            type      => "m.room.aliases",
+            state_key => $server_name,
+         )
+      })->then( sub {
+         my ( $body ) = @_;
+
+         assert_json_keys( $body, qw( aliases ) );
+         assert_json_list( my $aliases = $body->{aliases} );
+
+         not $room_alias ~~ @$aliases or die "Expected alias to not be in list";
```
Comment (leonerd) [inline tests/30rooms/05aliases.pl]: Ditto here with a call to `none`:

    none { $_ eq $room_alias } or
       die ...

File: tests/30rooms/05aliases.pl [hunk 5]
```diff
@@ -147,3 +147,185 @@
...
+            type      => "m.room.aliases",
+            state_key => $server_name,
+         )
+      })->then( sub {
+         my ( $body ) = @_;
+
+         assert_json_keys( $body, qw( aliases ) );
+         assert_json_list( my $aliases = $body->{aliases} );
+
+         $room_alias ~~ @$aliases or die "Expected alias to be in list";
```
Comment (leonerd) [inline tests/30rooms/05aliases.pl]: ditto

File: tests/30rooms/05aliases.pl [hunk 6]
```diff
@@ -147,3 +147,185 @@
...
+            type      => "m.room.aliases",
+            state_key => $server_name,
+         )
+      })->then( sub {
+         my ( $body ) = @_;
+
+         assert_json_keys( $body, qw( aliases ) );
+         assert_json_list( my $aliases = $body->{aliases} );
+
+         not $room_alias ~~ @$aliases or die "Expected alias to not be in list";
```
Comment (leonerd) [inline tests/30rooms/05aliases.pl]: ditto

**Comment by erikjohnston** [comment]
@leonerd PTAL

**Comment by leonerd** [comment]
LGTM asides one `=>` alignment issue still with a comment

**Comment by erikjohnston** [comment]
oh, i missed that one

**Pull request merged**